src/framework-ondata.js

```diff
@@ -456,3 +459,51 @@ function shouldRunUpdater(vni, isStale) {
     // TODO: May add different policies in the future.
     return (!isStale) && haveAllInputs(vni) && dataIsGood(vni);
 }
+
+/**
+ * Set default output metadata by appending input state metadata to the outputState 
+ * so it gets passed on to downstream nodes.
+ */
+function setDefaultMetadata(vni, inputState) {
+
+    var inputKeys = Object.keys(inputState);
+    var outputState = vni.outputState();
+    var outputKeys = Object.keys(outputState);
+
+    // Walk to list of keys looking for non-standard state keys - those will be metadata.
+    inputKeys.forEach(function(key) {
+        if (!_.contains(createState.STATE_KEYS, key)) {
+
+            // Found a metadata key - do we already know about it on the output? 
+            if (_.contains(outputKeys, key)) { 
+
+                // We already have an entry for this metadata key.  
+                var metadata = outputState[key];
+                if (_.isArray(metadata) && !_.contains(metadata, inputState[key])) {
+                    // Metadata is an array - push this value onto it with any earlier values
+                    // So downstream node can see them all
+                    logger.debug('Framework found multiple metadata values for ',key);
+                    metadata.push(inputState[key]);
+                    metadata = _.uniq(metadata);
+
+                } else if ((!_.isArray(metadata)) && (metadata !== inputState[key])) {
+                    // Metadata is currently a single value and we have two different values for it
+
+                    // Is either the new value or the old value undefined?  If so, keep whatever the new value is
+                    if (_.isUndefined(inputState[key]) || _.isUndefined(metadata)) { 
```

Comment:
I was deliberately allowing someone to reset a metadata value to undefined.  To do that, they would have to use the metadata property name and explicitly set the value for that property to undefined.  It would not happen if they simply don't have the property. 

@dbooth - can you weigh in on what you want functionally here?   Should the user be allowed to write something that deliberately clears an upstream metadata value?